app/services/HCAService.js

```diff
@@ -0,0 +1,264 @@
+    let authorities = [];
+
+    try {
+      const result = await this.fetchAuthoritiesYaml();
+      const list = await RNFetchBlob.fs.readFile(result.path(), 'utf8');
+      authorities = Yaml.safeLoad(list).Authorities;
+    } catch (err) {
+      console.log(err);
+    }
+
+    return authorities;
+  }
+
+  /**
+   * Get the list of Health Care Authorities that a user has saved
+   * @returns {Array} List of health care authorities from storage
+   */
+  async getUserAuthorityList() {
+    return await GetStoreData(AUTHORITY_SOURCE_SETTINGS, false);
+  }
+
+  /**
+   * Takes an array of one or more authorities and adds it to the list
+   * in storage that the user is subscribed to.
+   * @param {newAuthorities} Array healthcare authoritiy objects
+   * @returns void
+   */
+  async appendToAuthorityList(newAuthorities) {
+    const authorities = (await this.getUserAuthorityList()) || [];
+    await SetStoreData(AUTHORITY_SOURCE_SETTINGS, [
+      ...authorities,
+      ...newAuthorities,
+    ]);
+  }
+
+  /**
+   * Checks if a user has saved any Health Care Authorities
+   *
+   * @returns {boolean}
+   */
+  async hasSavedAuthorities() {
+    const authorities = await this.getUserAuthorityList();
+    return authorities && authorities.length > 0;
+  }
+
+  /**
+   * Alerts a user that there are new Healthcare Authorities in their region.
+   * Includes information on the number of Authorities in their current location.
+   *
+   * @param {numAuthorities} Number number of new authorities
+   * @returns {void}
+   */
+  async pushAlertNewAuthoritesFromLoc(numAuthorities) {
+    PushNotification.localNotification({
+      title: languages.t('label.authorities_new_in_area_title'),
+      message: languages.t('label.authorities_new_in_area_msg', {
+        count: numAuthorities,
+      }),
+    });
+  }
+
+  /**
+   * Alerts a user that they have been subscribed to new Healthcare Authorities
+   * in their region. Includes information on the number of Authorities in
+   * their current location.
+   *
+   * @param {numAuthorities} Number number of new authorities
+   * @returns {void}
+   */
+  async pushAlertNewSubscribedAuthorities(numAuthorities) {
+    PushNotification.localNotification({
+      title: languages.t('label.authorities_new_subcription_title', {
+        count: numAuthorities,
+      }),
+      message: languages.t('label.authorities_new_subcription_msg', {
+        count: numAuthorities,
+      }),
+    });
+  }
+
+  /**
+   * Returns the `url` value for an authority
+   * @param {authority} Authority Healthcare Authority object
+   * @returns {string}
+   */
+  getAuthorityUrl(authority) {
+    const authorityName = Object.keys(authority)[0];
+    const urlKey = authority[authorityName][0];
+    return urlKey && urlKey['url'];
+  }
+
+  /**
+   * Returns the `bounds` value for an authority
+   * @param {authority} Authority Healthcare Authority object
+   * @returns {Object} contains a `bounds` key.
```

Comment:
`@returns {{bounds: {... structure}}}`

until we get TypeScript, this actually does work with the IDE